### PR TITLE
Desktop support

### PR DESCRIPTION
**Pull request by ImmanuelBaskaran**

### CODE REVIEW

File: playground/theme-editor-showcase/src/App.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-import { useState, useEffect } from "react";
+import { useEffect, useState } from "react";
 import { BrowserRouter } from "react-router-dom";
 
 import { ThemeEditorApp } from "@jpmorganchase/theme-editor-app/src/ThemeEditorApp";
 import {
-  WindowContext,
   ElectronWindow,
+  WindowContext,
 } from "@jpmorganchase/uitk-lab/src/window";
```
Comment (joshwooding) [inline playground/theme-editor-showcase/src/App.tsx:9]: FYI

**Review by joshwooding — DISMISSED** [review]

File: packages/core/src/tooltip/useTooltip.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ import {
   useRef,
 } from "react";
 import { TooltipProps } from "./Tooltip";
+import { isDesktop } from "../window";
+import { margin } from "../utils/marginMiddleware";
```
Comment (joshwooding) [inline packages/core/src/tooltip/useTooltip.ts:26]: Was it intentional to not re-export this?

File: packages/lab/src/overlay/useOverlay.ts [hunk 3]
```diff
@@ -23,6 +24,8 @@ import {
 } from "react";
 import { OverlayProps } from "./Overlay";
 
+import { margin } from "@jpmorganchase/uitk-core/src/utils/marginMiddleware";
```
Comment (joshwooding) [inline packages/lab/src/overlay/useOverlay.ts:27]: This worries me.

File: packages/lab/src/combo-box/internal/MultiSelectComboBox.tsx [hunk 4]
```diff
@@ -100,19 +101,22 @@ export function MultiSelectComboBox<Item>(
...
-        fallbackPlacements: ["bottom-start", "top-start"],
-      }),
-      shift({ limiter: limitShift() }),
-      size({
-        apply({ availableHeight }) {
-          setMaxListHeight(availableHeight);
-        },
-      }),
-    ],
+    middleware: middleware,
```
Comment (alycrys) [inline packages/lab/src/combo-box/internal/MultiSelectComboBox.tsx:119]: just `middleware` suffice here

File: packages/lab/src/dropdown/useDropdown.ts [hunk 5]
```diff
@@ -234,7 +234,7 @@ export function useDropdown<Item, Variant extends ListSelectionVariant>(
   };
 
   const handleButtonBlur = (event: FocusEvent<HTMLDivElement>) => {
-    // Next line causes an issue for DesktopWindows. Need to consider how we fix.
+    // Next line causes an issue for ElectronWindow. Need to consider how we fix.
```
Comment (alycrys) [inline packages/lab/src/dropdown/useDropdown.ts:237]: maybe add a TODO also?

File: packages/lab/src/window/ElectronWindow.tsx [hunk 6]
```diff
@@ -7,85 +8,118 @@ import {
...
+  useEffect(() => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
     const { ipcRenderer } = global as any;
     if (ipcRenderer) {
-      ipcRenderer.send("window-ready", { id });
+      setTimeout(() => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
+        ipcRenderer.send("window-ready", { id: id });
+      }, 100);
     }
```
Comment (alycrys) [inline packages/lab/src/window/ElectronWindow.tsx:80]: There must be a good reason for this 100ms timeout. Maybe it's worth a comment

File: packages/core/src/utils/marginMiddleware.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { Middleware } from "@floating-ui/core";
+import { MiddlewareArguments } from "@floating-ui/react-dom-interactions";
+
+export const margin = (value = 0): Middleware => ({
```
[the remark this answers is not in the document]
Comment (ImmanuelBaskaran) [reply packages/core/src/utils/marginMiddleware.ts:4]: I was following the conventions of the other middleware created by floating UI

**Review by heswell — APPROVED** [review]

**Pull request merged**